src/lib/logical_query_plan/aggregate_node.cpp

```diff
@@ -77,6 +77,48 @@ std::vector<std::shared_ptr<AbstractExpression>> AggregateNode::output_expressio
   return output_expressions;
 }
 
+std::shared_ptr<LQPUniqueConstraints> AggregateNode::unique_constraints() const {
+  auto unique_constraints = std::make_shared<LQPUniqueConstraints>();
+
+  // (1) Create new unique constraint from the group-by column(s) which form a candidate key for the output relation.
+  const auto group_by_columns_count = aggregate_expressions_begin_idx;
+  ExpressionUnorderedSet group_by_columns(group_by_columns_count);
+  std::copy_n(node_expressions.begin(), group_by_columns_count,
+              std::inserter(group_by_columns, group_by_columns.begin()));
+
+  unique_constraints->emplace_back(group_by_columns);
+
+  // (2) Forward input constraints, if applicable
+  // We call output_expressions() to avoid ANY() aggregates from the DependentGroupByReductionRule
+  const auto& output_expressions_vec = this->output_expressions();
+  const auto output_expressions =
+      ExpressionUnorderedSet{output_expressions_vec.cbegin(), output_expressions_vec.cend()};
+
```

Comment:
Yes and no. Currently, we would lose the unique constraint if one of its columns becomes ANY()-wrapped.

However, by using `AggregateNode::output_expressions()` ANY() expressions get unwrapped. Therefore, `output_expressions.contains(expression)` returns true and we do not lose the constraint.


I changed the comment to 

``` cpp
// The DependentGroupByReductionRule might wrap some expressions within ANY() aggregate functions.
// However, ANY() acts as a pure marker for optimizations and does not change the expressions themselves. Therefore, 
// unique constraints from child nodes should not be discarded because of ANY() expressions.
// To simplify the forwarding logic, we call output_expressions() to get all expressions without the ANY() wrapping.
```

Reply:
Having a hard time expressing this.

If we have a table with columns `a,b,c,d` and FDs `b->c` and group on `a,b,c`, the reduction would change it to group on `a,b` and add `c` as an ANY column. One might expect the output unique constraint `a,b,c`, but it is (correctly) only `a,b`. That is good, because the fewer columns are included, the "better" the constraint is. I think that this behavior should be doc'ed.

Also, `a,c` would be a possible output constraint. Just like with the other place where we do not yet generate all possible constraints, we should hint at this possible extension.

Reply:
Very good idea! Have not thought about this yet. I'll try to incorporate it.

Reply:
Alright, I had my first run on this and I think we should address this in a separate PR. 

Your suggestion might lead to a general FD-driven optimization function and several tests.
I have thought about different scenarios, which led to several new questions. E.g.:
* How to handle optimized AggregateNodes vs. non-optimized AggregateNodes (DependentGroupByReductionRule)
* How to handle other FDs, such as {b} => {c, d}, {a, b} => {c, d}, ...
* Ordering of FDs for best optimization results

--- 

Apart from that, the current implementation creates valid unique constraints and profits from the DependentGroupByReductionRule in case it has been executed beforehand.

Reply:
> Alright, I had my first run on this and I think we should address this in a separate PR.

I think I'd just doc this discussion for now and hint at the possible optimization.